pkg/parse/root.go

```diff
@@ -261,6 +261,38 @@ func setSourceStatusFields(source *v1beta1.SourceStatus, newStatus *SourceStatus
 	source.LastUpdate = newStatus.LastUpdate
 }
 
+func (p *root) setSourceAnnotations(ctx context.Context, commit string) error {
+	rs := &v1beta1.RootSync{}
+	if err := p.Client.Get(ctx, rootsync.ObjectKey(p.SyncName), rs); err != nil {
```

Comment:
Is the get necessary? Seems like this patch does not require a get

Reply:
It compares the annotations with the new values to avoid unnecessary patching.

Reply:
Alternatively, instead of merging from existing object, we can patch directly with the raw patch data, `client.RawPatch`. With that, we don't need to GET and check for changes.